sycl/source/detail/device_info.hpp

```diff
@@ -568,6 +568,47 @@ struct get_device_info_impl<range<Dimensions>,
   }
 };
 
+#define ARCHES(X)                                                              \
+  X("5.0", oneapi_exp_arch::nvidia_gpu_sm_50)                                  \
```

Comment:
Is it possible to unify the code and make it the same for Intel GPUs and for AMD/NVIDIA GPUs? E.g., use everywhere this macro or use everywhere std::map with these values

Reply:
It is. I've converted the original deviceIP values inside the std::map to std::string and moved the pairs inside std::map to the X macro.